[2019.2.1] Various fixes for 2019.2.1 to ensure tests pass on Mac OS X

garethgreenaway opened the pull request:
### What does this PR do?
Various fixes for 2019.2.1 to ensure tests pass on Mac OS X

### What issues does this PR fix or reference?
N/A

### Tests written?
**[NOTICE] Bug fixes or features added to Salt require tests.**
Please review the [test documentation](https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite.
Yes.  Fixes to existing tests.

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

garethgreenaway committed:
Fixing the beacons.reset function.  Once the reset has taken place in beacons/__init__.py we need to fire an event back to complete the loop and ensure that everything worked as expected.

garethgreenaway committed:
Fixing a bug when the roots fileserver and the location is a symlink to another location.  This fix ensures that when fsroot is referenced we are using the real path and not the symlink path.

garethgreenaway committed:
Fixing a log issue that pops up after test_gen_thin_compression_fallback_py3 on "OS X, need to ensure that salt.utils.thin.os.close is mocked.

garethgreenaway committed:
fix to how the depends decorator works.  Only run the dependancy commands for the module we're checking.

garethgreenaway committed:
Format for the sqlite3 databse used for the assistive information changed in Mojave, additional columns added.

weswhet commented:
LGTM: Thanks for taking the time to give macOS a little TLC.

s0undt3ch commented:
The http query test failures under py3 will be fixed by https://github.com/saltstack/salt/commit/b1b1b265ebe25338ec073ccdf5c2a9522d0ff671 https://github.com/saltstack/salt/pull/53408

garethgreenaway committed:
Using salt.utils.path.which to find false, on OSX it lives under /usr/bin/

garethgreenaway committed:
Ensure the user and group used by @with_system_user_and_group are consistent.

garethgreenaway committed:
Adding some timeouts to see if it helps the tests pass on OS X.

garethgreenaway committed:
Disabling beacon state tests on OS X for the time being.

garethgreenaway committed:
Moving the skipIf for OS X from the beacon state tests to the renderer state tests.

garethgreenaway committed:
Removing wrong import of skipIf

garethgreenaway committed:
Skip integration.states.test_file.FileTest.test_issue_11003_immutable_lazy_proxy_sum and
integration.states.test_pip_state.PipStateTest.test_22359_pip_installed_unless_does_not_trigger_warnings when running on OS X.

garethgreenaway committed:
Disabling integration.modules.test_state.StateModuleTest.test_parallel_state_with_long_tag

garethgreenaway committed:
Skipping integration.modules.test_cp.CPModuleTest.test_get_file_str_https on OS X

garethgreenaway committed:
Updating various skips for tests to only skip if OS is OS X and Python is Python2.

garethgreenaway committed:
Fixing lint.

garethgreenaway committed:
Disabling a couple more tests in integration.modules.test_cp that hang on OS X and Py2.
